scripts/commands/goto.lua

```diff
@@ -17,32 +17,27 @@ end;
 function onTrigger(player, target, forceZone)
 
     -- validate target
-    if (target == nil) then
+    if not target then
         error(player, "You must enter a player name.");
-        return;
-    end
-    local targ = GetPlayerByName( target );
-    if (targ == nil) then
-        if not player:gotoPlayer( target ) then
-            error(player, string.format( "Player named '%s' not found!", target ) );
-        end
-        return;
+        return
     end
 
     -- validate forceZone
-    if (forceZone ~= nil) then
-        if (forceZone ~= 0 and forceZone ~= 1) then
-            error(player, "If provided, forceZone must be 1 (true) or 0 (false).");
-            return;
+    if forceZone then
+        if forceZone ~= 0 and forceZone ~= 1 then
+            error(player, "If provided, forceZone must be 1 (true) or 0 (false).")
+            return
         end
     else
-        forceZone = 1;
+        forceZone = 1
     end
 
-    -- goto target
-    if (targ:getZoneID() ~= player:getZoneID() or forceZone == 1) then
-        player:setPos( targ:getXPos(), targ:getYPos(), targ:getZPos(), targ:getRotPos(), targ:getZoneID() );
-    else
-        player:setPos( targ:getXPos(), targ:getYPos(), targ:getZPos(), targ:getRotPos() );
+    local targ = GetPlayerByName(target)
+    -- if we found this player, they're on the same zone server
+    -- if they're in mog house, goto them instead of setPos
+    if targ and not targ:isInMogHouse() then
+        player:setPos(targ:getXPos(), targ:getYPos(), targ:getZPos(), targ:getRotPos(), forceZone == 1 and targ:getZoneID() or nil)
+    elseif not player:gotoPlayer(target) then
```

Comment:
Why not just always used `gotoPlayer()`?

Reply:
because someone added `forceZone` param to prevent zoning, only setPos if within same zone and `gotoPlayer` just uses server message to handle everything

Reply:
oooh right. goto can't move pos in same zone without a rezone. forgot.